projects/ngx-charts-on-fhir/src/lib/fhir-data/fhir-data.service.ts

```diff
@@ -95,15 +98,47 @@ export class FhirDataService {
       // use maximum page size on every request to improve performance
       url = addCountParam(url);
       const request = currentPatientOnly ? this.client.patient.request.bind(this.client.patient) : this.client.request.bind(this.client);
+
       request(url, { pageLimit: 0, onPage })
         .then(() => subscriber.complete())
         .catch((error) => subscriber.error(error));
       return teardownLogic;
-    });
+    }).pipe(
+      retryBackoff({
+        initialInterval: this.INITIAL_INTERVAL,
+        maxRetries: this.MAX_RETRIES,
+        resetOnSuccess: true,
+        shouldRetry: (error) => {
+          return error.status > 500 && error.status < 505;
+        },
+      })
+    );
   }
 
-  addPatientData(resource: Resource) {
-    return this.client?.create(resource);
+
+  addPatientData<R extends Resource>(resource: R): Observable<R> {
+    return new Observable<R>((subscriber) => {
+      if (!this.client) {
+        subscriber.error('FhirClientService has not been initialized.');
+        return;
+      }
+      this.client
+        .create<R>(resource)
+        .then((res) => {
+          subscriber.next(res);
+          subscriber.complete();
+        })
+        .catch((error) => subscriber.error(error));
+    }).pipe(
+      retryBackoff({
+        initialInterval: this.INITIAL_INTERVAL,
+        maxRetries: this.MAX_RETRIES,
+        resetOnSuccess: true,
+        shouldRetry: (error) => {
+          return error.status > 500 && error.status < 505;
+        },
```

Comment:
this will not retry 500
```suggestion
        shouldRetry: (error) => {
          return error.status >= 500 && error.status < 505;
        },
```